plugins/modules/packaging/os/homebrew_cask.py

```diff
@@ -454,18 +464,40 @@ def _current_cask_is_installed(self):
             self.message = 'Invalid cask: {0}.'.format(self.current_cask)
             raise HomebrewCaskException(self.message)
 
-        cmd = [
-            "{brew_path}".format(brew_path=self.brew_path),
-            "cask",
-            "list",
-            self.current_cask
-        ]
+        if self._brew_cask_command_is_deprecated():
+            base_opts = [self.brew_path, "list", "--cask"]
+        else:
+            base_opts = [self.brew_path, "cask", "list"]
+
+        cmd = base_opts + [self.current_cask]
         rc, out, err = self.module.run_command(cmd)
 
         if rc == 0:
             return True
         else:
             return False
+
+    def _get_brew_version(self):
+        if self.brew_version:
+            return self.brew_version
+
+        cmd = [self.brew_path, '--version']
+
+        rc, out, err = self.module.run_command(cmd, check_rc=True)
+
+        if rc == 0:
```

Comment:
If `check_rc=True` is passed, `rc` will always be 0.

Reply:
not 100% sure how the `check_rc` is meant to be used. in this case, should i omit `check_rc=True` and let it default to `False`?

Reply:
If `check_rc=True` is specified, `run_command` will fail if `rc != 0`, and only return if `rc == 0`. So you don't need the whole `if`.

Reply:
i see. what about error handling? 
does that mean if `rc != 0`, i need not explicitly parse the error message and throw `HomebrewCaskException`?

Reply:
The relevant code inside `run_command()` goes like this:
```python
        if rc != 0 and check_rc:
            msg = heuristic_log_sanitize(stderr.rstrip(), self.no_log_values)
            self.fail_json(cmd=self._clean_args(args), rc=rc, stdout=stdout, stderr=stderr, msg=msg)
```
So as you can see, stderr output will be sent to the outside. If that is good enough for you, then the answer to your question is "yes, it is sufficient". If that does not work, or if you want to do something different, you can pass `check_rc=False` and do the error handling in any way you want.

Reply:
What `run_command()` does is totally sufficient here.

Reply:
thank you both for clarifying @russoz @felixfontein . i have made the relevant changes